buildkite/aggregate_incompatible_flags_test_result.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+#
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+import json
+import os
+import re
+import subprocess
+import sys
+import threading
+import yaml
+import urllib.request
+
+import bazelci
+from bazelci import BuildkiteException
+
+BUILD_STATUS_API_URL = "https://api.buildkite.com/v2/organizations/bazel/pipelines/bazelisk-plus-incompatible-flags/builds/"
+
+ENCRYPTED_BUILDKITE_API_TOKEN = """
+CiQA4DEB9ldzC+E39KomywtqXfaQ86hhulgeDsicds2BuvbCYzsSUAAqwcvXZPh9IMWlwWh94J2F
+exosKKaWB0tSRJiPKnv2NPDfEqGul0ZwVjtWeASpugwxxKeLhFhPMcgHMPfndH6j2GEIY6nkKRbP
+uwoRMCwe
+""".strip()
+
+INCOMPATIBLE_FLAGS = bazelci.fetch_incompatible_flags()
+
+class LogFetcher(threading.Thread):
+   def __init__(self, job):
+      threading.Thread.__init__(self)
+      self.job = job
+      self.log = None
+
+   def run(self):
+      self.log = get_build_log(self.job)
+
+
+def buildkite_token():
+    return (
+        subprocess.check_output(
+            [
+                bazelci.gcloud_command(),
+                "kms",
+                "decrypt",
+                "--project",
+                "bazel-untrusted",
+                "--location",
+                "global",
+                "--keyring",
+                "buildkite",
+                "--key",
+                "buildkite-untrusted-api-token",
+                "--ciphertext-file",
+                "-",
+                "--plaintext-file",
+                "-",
+            ],
+            input=base64.b64decode(ENCRYPTED_BUILDKITE_API_TOKEN),
+            env=os.environ,
+        )
+        .decode("utf-8")
+        .strip()
+    )
+
+
+def get_build_status_api_url(build_number):
+    return BUILD_STATUS_API_URL + "%s?access_token=%s" % (build_number, buildkite_token())
+
+
+def get_build_log_api_url(raw_log_url):
+    return raw_log_url + "?access_token=" + buildkite_token()
+
+
+def get_build_info(build_number):
+    output = urllib.request.urlopen(get_build_status_api_url(build_number)).read().decode("utf-8")
+    build_info = json.loads(output)
+    return build_info
+
+
+def get_build_log(job):
+    log_url = job["raw_log_url"]
+    output = urllib.request.urlopen(get_build_log_api_url(log_url)).read().decode("utf-8")
+    return output
+
+
+def process_build_log(failed_jobs_per_flag, log, job):
+    if log.find("Success: No migration needed.") != -1:
```

Comment:
why not just 
if "Success: No migration needed." in log:

(and below).

Reply:
Done, thanks!